### PR TITLE
Requesting a pull to mozilla:master from k88hudson:t2113b

### PR DESCRIPTION
**Pull request by k88hudson**

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @secretrobotron received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/13ede05b9132e6a/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @secretrobotron received. Current queue size: 2

Live output at: http://botio.butyr.org:8000/3893458232fbaf7/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/13ede05b9132e6a/output.txt

Total script time: 3.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/3893458232fbaf7/output.txt

Total script time: 3.00 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_cleanup** from @secretrobotron received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/1f89af98881e437/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/1f89af98881e437/output.txt

Total script time: 0.90 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @secretrobotron received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/1eb72c1e1d74d64/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/1eb72c1e1d74d64/output.txt

Total script time: 2.30 mins

File: src/core/views/trackevent-view.js [hunk 1]
```diff
@@ -30,19 +30,17 @@ define( [ "core/logger", "core/eventmanager", "util/dragndrop",
...
-      }
-    }
-
     function resetContainer() {
       if ( !_trackEvent.track || !_trackEvent.track._media ) {
         return;
       }
       _element.style.left = _start  / _trackEvent.track._media.duration * 100 + "%";
       _element.style.width = ( _end - _start ) / _trackEvent.track._media.duration * 100 + "%";
+      if ( _element.getBoundingClientRect().width < 50 ) {
```
Comment (secretrobotron) [inline src/core/views/trackevent-view.js]: Replace `50` with a const, please.

File: css/tray-timeline.less [hunk 2]
```diff
@@ -122,26 +131,65 @@
...
+    width: 18px;
+    height: 18px;
+    top: 5px;
+    background: #FFF;
+    &:after {
+      opacity: 0.2;
+      content: "";
+      position: absolute;
+      display: inline-block;
+      background-image: url("../resources/glyphicons-halflings-green.png");
```
Comment (secretrobotron) [inline css/tray-timeline.less]: Spacing.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @secretrobotron received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/0ce35332c0fa923/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @secretrobotron received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/c7fc45d66311fea/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/0ce35332c0fa923/output.txt

Total script time: 1.64 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/c7fc45d66311fea/output.txt

Total script time: 1.32 mins

Server is running on http://pr984.butyr.org/templates/basic/
Run the Test Suite: http://pr984.butyr.org/test/

File: css/tray-timeline.less [hunk 3]
```diff
@@ -124,24 +132,72 @@
...
-    height: 100%;
-    min-width: 15px;
-    width: 15px;
-    background-color: rgba( 255, 255, 255, 0.5 );
+    height: 18px;
+    width: 18px;
+    top: 5px;
+    opacity: 0;
+    .transition( all 0.1s ease );
+    background: #FFF;
```
Comment (secretrobotron) [inline css/tray-timeline.less]: Do you want to base this off of something else, or make the colour a global somewhere? Seems like something tweakable enough that fishing through LESS/CSS would be a pain.

File: css/tray-timeline.less [hunk 4]
```diff
@@ -124,24 +132,72 @@
...
+    &:after {
+      content: "";
+      position: absolute;
+      display: inline-block;
+      background-image: url("../resources/glyphicons-halflings-green.png");
+      width: 14px;
+      height: 14px;
+      top: 2px;
+
```
Comment (secretrobotron) [inline css/tray-timeline.less]: Extra spacing.

File: external/popcorn-js [hunk 5]
```diff
@@ -1 +1 @@
-Subproject commit 0cd8d1322c0b0a9e400877ccc68960e1f88d2a0f
+Subproject commit b6595fd07dd4a4db01c8565a4f2167cec80e47b8
```
Comment (secretrobotron) [inline external/popcorn-js]: I don't think you meant to do this, right? If you can avoid it, don't update the Popcorn submodule incidentally.

File: external/popcorn-js [hunk 6]
```diff
@@ -1 +1 @@
-Subproject commit b6595fd07dd4a4db01c8565a4f2167cec80e47b8
+Subproject commit 0cd8d1322c0b0a9e400877ccc68960e1f88d2a0f
```
Comment (secretrobotron) [inline external/popcorn-js]: This is still wrong. Did you rebase?

File: src/core/views/trackevent-view.js [hunk 7]
```diff
@@ -7,6 +7,8 @@ define( [ "core/logger", "core/eventmanager", "util/dragndrop",
   function( Logger, EventManagerWrapper, DragNDrop,
             LangUtils, TRACKEVENT_LAYOUT ) {
 
+  var TRACKEVENT_MIN_WIDTH = 50;
+
```
Comment (secretrobotron) [inline src/core/views/trackevent-view.js]: whitespace

File: src/util/dragndrop.js [hunk 8]
```diff
@@ -620,6 +620,7 @@ define([], function(){
       // Make sure the position is up to date after this call because the user may
       // have moved the element around in the DOM tree.
       _draggable.updateRects();
+      element.classList.add( "butter-dragging" );
```
Comment (secretrobotron) [inline src/util/dragndrop.js]: This is the right idea, but it's in the wrong spot. `Resizable` is what needs patching.

File: css/tray-timeline.less [hunk 9]
```diff
@@ -124,24 +134,78 @@
...
+    width: 18px;
+    top: 5px;
+    opacity: 0;
+    background: @_trackEventColor;
+    .transition( all 0.1s ease );
+    &:after {
+      content: "";
+      position: absolute;
+      display: inline-block;
+      background-image: url("../resources/glyphicons-halflings-green.png");
```
Comment (secretrobotron) [inline css/tray-timeline.less]: Spacing.

File: src/util/dragndrop.js [hunk 10]
```diff
@@ -181,7 +181,8 @@ define([], function(){
   }
 
   function Resizable( element, options ){
-    var _leftHandle = element.querySelector( ".handle.left-handle" ),
+    var _draggingClass = "butter-dragging",
```
Comment (secretrobotron) [inline src/util/dragndrop.js]: This should be a const, not a member variable. Probably `DRAGGING_CLASS` near the top of the file.

Comment (secretrobotron) [inline src/util/dragndrop.js]: Also, `butter-dragging` is no longer appropriate. Perhaps `butter-resizing`.

File: src/util/dragndrop.js [hunk 11]
```diff
@@ -246,6 +248,7 @@ define([], function(){
         }
       }
 
+      element.classList.add( _draggingClass );
```
Comment (secretrobotron) [inline src/util/dragndrop.js]: This is great, but you need to worry about the right side too. This code only works for the left.

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_check** from @mjschranz received. Current queue size: 1

Live output at: http://botio.butyr.org:8000/c2c7ca6617ee18e/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @mjschranz received. Current queue size: 2

Live output at: http://botio.butyr.org:8000/06da7a06be3bef2/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/c2c7ca6617ee18e/output.txt

Total script time: 1.74 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Success

Full output at http://botio.butyr.org:8000/06da7a06be3bef2/output.txt

Total script time: 1.16 mins

Server is running on http://pr984.butyr.org/templates/basic/
Run the Test Suite: http://pr984.butyr.org/test/

File: css/tray-timeline.less [hunk 12]
```diff
@@ -124,24 +134,73 @@
...
+    .transition( all 0.1s ease );
+    &:after {
+      content: "";
+      position: absolute;
+      display: inline-block;
+      background-image: url( "../resources/glyphicons-halflings-green.png" );
+      width: 14px;
+      height: 14px;
+      top: 2px;
+
```
Comment (mjschranz) [inline css/tray-timeline.less]: Remove the new line.

File: src/core/views/trackevent-view.js [hunk 13]
```diff
@@ -262,12 +261,10 @@ define( [ "core/logger", "core/eventmanager", "util/dragndrop",
                 _handles = _element.querySelectorAll( ".handle" );
                 if( _handles && _handles.length === 2 ){
                   _element.addEventListener( "mouseover", function( e ){
-                    toggleHandles( true );
                   }, false );
                   _element.addEventListener( "mouseout", function( e ){
-                    toggleHandles( false );
                   }, false );
-                  toggleHandles( false );
+
```
Comment (mjschranz) [inline src/core/views/trackevent-view.js]: These event listeners aren't actually doing anything now. Could you remove them and the entire if blocks around them?

**Pull request merged**